# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt
import datetime

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
# Checking the original data type
df.dtypes

station     object
date        object
prcp       float64
tobs         int64
dtype: object

In [4]:
# Convert the date column format from string to datetime

df["month"] = pd.to_datetime(df["date"]).dt.strftime("%d")
df["date"] = pd.to_datetime(df["date"])
df.dtypes


station            object
date       datetime64[ns]
prcp              float64
tobs                int64
month              object
dtype: object

In [5]:
# Set the date column as the DataFrame index
# Drop the date column

df.set_index('date',inplace=True)
df.head()

station  prcp  tobs month
date                                     
2010-01-01  USC00519397  0.08    65    01
2010-01-02  USC00519397  0.00    63    02
2010-01-03  USC00519397  0.00    74    03
2010-01-04  USC00519397  0.00    76    04
2010-01-06  USC00519397   NaN    73    06

### Compare June and December data across all years 

In [6]:
from scipy import stats

In [7]:
# Filter data for desired months (JUN)
df_filtered = df.loc[df["month"]=="06"]
df_filtered.head()

station  prcp  tobs month
date                                     
2010-01-06  USC00519397   NaN    73    06
2010-02-06  USC00519397  0.00    67    06
2010-03-06  USC00519397  0.03    72    06
2010-04-06  USC00519397  0.01    75    06
2010-05-06  USC00519397  0.00    80    06

In [8]:
# Identify the average temperature for June
df_filtered = df.loc[df["month"]=="06"]
jun_avg = df_filtered["tobs"].mean()
jun_avg

72.7361963190184

In [9]:
# Identify the average temperature for December
df_filtered = df.loc[df["month"]=="12"]
dec_avg = df_filtered["tobs"].mean()
dec_avg

72.91833590138675

In [10]:
# Create collections of temperature data
df_jun= df.loc[df["month"]=="06"]
df_dec= df.loc[df["month"]=="12"]


In [11]:
jun=df_jun["tobs"]
dec=df_dec["tobs"]

In [12]:
# Run paired t-test
import numpy as np
from scipy import stats

print(np.var(jun), np.var(dec))
print(np.var(dec)/np.var(jun))

res=stats.ttest_ind(jun, dec,equal_var=True)
display(res)

18.896665286612137 22.296874888711134
1.1799370180149176


Ttest_indResult(statistic=-0.7233017834086564, pvalue=0.4696247146533662)

### Analysis

Since the p-value (0.4696247146533662) is less than 0.05, there is an evidence to say mean of jun and dec temperatures are different.